### PR TITLE
pr-builder workflow

### PR DESCRIPTION
**Pull request by PasinduLakshan99**
### integrate pr-builder workflow

### CODE REVIEW

File: .github/workflows/pr-builder.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This workflow will build the project on pull requests with tests
+# Uses:
+#   OS: ubuntu-latest
+#   JDK: Adopt JDK 8
```
Comment (Thumimku) [inline .github/workflows/pr-builder.yml:4]: version?

File: .github/workflows/pr-builder.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+env:
+  MAVEN_OPTS: -Xmx4g -Xms1g
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Adopt JDK 8
```
Comment (Thumimku) [inline .github/workflows/pr-builder.yml:22]: Name?

File: .github/workflows/pr-builder.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        env:
+          cache-name: cache-m2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ env.cache-name }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-${{ env.cache-name }}-
+            ${{ runner.os }}-maven-
+            ${{ runner.os }}-
+      - name: Build with Maven
```
Comment (Thumimku) [inline .github/workflows/pr-builder.yml:39]: Delete .m2 after building

Comment (Thumimku) [reply .github/workflows/pr-builder.yml:39]: Let's delete the snapshot artifacts only refer: https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/.github/workflows/pr-builder.yml

File: .github/workflows/pr-builder.yml (hunk 2 above)
Comment (PasinduLakshan99) [reply .github/workflows/pr-builder.yml]: fixed with https://github.com/wso2-extensions/identity-oauth-dpop/pull/5/commits/d213e4203f00f4eccfca4db70b07e55a1c4ae5ca

File: .github/workflows/pr-builder.yml (hunk 1 above)
Comment (PasinduLakshan99) [reply .github/workflows/pr-builder.yml]: fixed with https://github.com/wso2-extensions/identity-oauth-dpop/commit/d213e4203f00f4eccfca4db70b07e55a1c4ae5ca

File: .github/workflows/pr-builder.yml (hunk 3 above)
Comment (PasinduLakshan99) [reply .github/workflows/pr-builder.yml:39]: fixed with https://github.com/wso2-extensions/identity-oauth-dpop/commit/d213e4203f00f4eccfca4db70b07e55a1c4ae5ca

**Pull request merged**